### DISCUSSION TITLE
Favorite songs should like their backend service

### DISCUSSION THREAD

**Discussion by eimirae** [music-assistant/support#2164] [Feature requests and ideas]
When favoriting a song, the song provider should also like the song (if available).
I'm not sure how to handle when a song is available on multiple providers? Perhaps favorite them all?

Ideally, also, make favorite buttons available on the player.

Thanks for considering this!

**Comment by OzGav** [comment]
That should already be happening. What provider did you use?

**Comment by eimirae** [comment]
Hi, sorry for the late response,

I'm using ytmusic.

I've been searching around the source code,

I believe the function used within the ytmusicapi would be rate_song: https://ytmusicapi.readthedocs.io/en/stable/reference.html#ytmusicapi.YTMusic.rate_song

but searching in the source code for the helpers and ytmusic provider, I can find rate_playlist references: https://github.com/music-assistant/server/blob/bf9b16dc61c558bf784f6e13a55cf6e5eecff7c4/music_assistant/server/providers/ytmusic/helpers.py#L179

But no rate_song function calls.

This matches my experience of using the program, where I can like a playlist and see that reflected within youtube, but liking a song does not cause youtube to have a corresponding like.

**Reply by OzGav** [reply]
@MarvinSchenkel ?

**Reply by MarvinSchenkel** [reply]
Sounds like something I missed! Could you make an issue with this request so I can assign it to myself?

**Reply by OzGav** [reply]
I’ll push it to an issue from here

**Reply by eimirae** [reply]
First off all, thanks for this integration, its great.

Second of all, I did some digging on what is happening, and here is what I found if it is helpful at all.

https://github.com/music-assistant/server/blob/bf9b16dc61c558bf784f6e13a55cf6e5eecff7c4/music_assistant/server/controllers/music.py#L437
Indicates that LIBRARY_TRACKS_EDIT feature is needed, which might possibly be a deeper can of worms for other functions behavior based on this feature.

**Reply by MarvinSchenkel** [reply]
It's basically the other way around. Adding that feature indicates to MA that the provider supports adding tracks to the remote library. I just need to implement it for YTM.

**Comment by OzGav** [comment]
sync and import options will be added to the 2.7 beta

**Reply by YoshiWalsh** [reply]
Hi, I'm using 2.8.7. Favouriting a song in MASS doesn't mark it as liked in YouTube Music. Is there a setting I have to enable somewhere to make this work, or should I submit a bug report?

**Reply by OzGav** [reply]
Looking at the YTM code it is not a bug as it is not supported.
